manifests/init.pp

```diff
@@ -49,28 +55,45 @@
     default => $user
   }
 
+  $composer_group = $group ? {
+    'UNDEF' => $::composer::params::group,
+    default => $group
+  }
+
+  # common dependency
+  package { 'patch':
+    ensure => latest
+  }
+
+
+  # fetch composer into target_dir
   wget::fetch { 'composer-install':
     source      => $::composer::params::phar_location,
     destination => "${composer_target_dir}/${composer_command_name}",
-    execuser    => $composer_user,
+    execuser    => 'root',
   }
 
-  exec { 'composer-fix-permissions':
-    command => "chmod a+x ${composer_command_name}",
-    path    => '/usr/bin:/bin:/usr/sbin:/sbin',
-    cwd     => $composer_target_dir,
-    user    => $composer_user,
-    unless  => "test -x ${composer_target_dir}/${composer_command_name}",
+
+  # apply user and group permissions to downloaded composer using native puppet File type
+  file { 'composer-fix-permissions':
+    ensure  => present,
+    path    => "${composer_target_dir}/${composer_command_name}",
+    owner   => $composer_user,
+    group   => $composer_group,
+    mode    => 'a+rx',
+    recurse => false,
     require => Wget::Fetch['composer-install'],
   }
 
+  # run self update when requested
   if $auto_update {
     exec { 'composer-update':
-      command     => "${composer_command_name} self-update",
-      environment => [ "COMPOSER_HOME=${composer_target_dir}" ],
+      command     => "php -d allow_url_fopen=1 ${composer_target_dir}/${composer_command_name} self-update",
```

Comment:
does this mean that it is not possible to access the composer by its command name??
 If it is possible then you should keep the old code. If not then I'm :-1: